vendor/src/github.com/docker/swarmkit/api/ca.proto

```diff
@@ -43,6 +43,9 @@ message IssueNodeCertificateRequest {
 	// Token represents a user-provided string that is necessary for new
 	// nodes to join the cluster
 	string token = 3;
+
+	// Availability allows a user to control the current scheduling status of a node.
+	string availability = 4;
```

Comment:
@yongtang: It sounds like the consensus is that we should provide a `NodeSpec` instead of just an availability setting. But as I mentioned earlier, the node CA should only use certain fields from this spec when it creates the node. There are things like node labels that are controlled from elsewhere.